contracts/cross-chain/ZNSZChainPortal.sol

```diff
@@ -0,0 +1,221 @@
+            || address(znsContracts.treasury) == address(0)
+            || address(znsContracts.chainResolver) == address(0)
+            || address(znsContracts.registry) == address(0)
+        ) revert ZeroAddressPassed();
+
+        polygonZkEVMBridge = zkEvmBridge_;
+        destNetworkId = destNetworkId_;
+        destChainName = destChainName_;
+        destChainId = destChainId_;
+        rootRegistrar = znsContracts.rootRegistrar;
+        subRegistrar = znsContracts.subRegistrar;
+        treasury = znsContracts.treasury;
+        chainResolver = znsContracts.chainResolver;
+        registry = znsContracts.registry;
+    }
+
+    function registerAndBridgeDomain(
+        bytes32 parentHash,
+        string calldata label,
+        string calldata tokenURI
+//      bool forceUpdateGlobalExitRoot  - do we need to pass this ???
+    ) external override {
+        DistributionConfig memory emptyDistrConfig;
+        PaymentConfig memory emptyPaymentConfig;
+
+        _processPayment(label, parentHash);
+
+        // Register domain
+        bytes32 domainHash;
+        if (parentHash == bytes32(0)) { // 0x0 parent for root domains
+            domainHash = rootRegistrar.registerRootDomain(
+                label,
+                address(0),
+                tokenURI,
+                emptyDistrConfig,
+                emptyPaymentConfig
+            );
+        } else {
+            domainHash = subRegistrar.registerSubdomain(
+                parentHash,
+                label,
+                address(0),
+                tokenURI,
+                emptyDistrConfig,
+                emptyPaymentConfig
+            );
+        }
+
+        // TODO multi: analyze how to make these strings better !!!
+        registry.updateDomainResolver(domainHash, "chain");
+        // TODO multi: iron out what should go to ChainResolver data !!!
+        chainResolver.setChainData(
+            domainHash,
+            destChainId,
+            destChainName,
+            address(0),
+            ""
+        );
+
+        // Bridge domain
+        _bridgeDomain(
+            domainHash,
+            parentHash,
+            label,
+            tokenURI
+        );
+    }
+
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes calldata
+    ) external pure override returns (bytes4) {
+        return this.onERC721Received.selector;
+    }
+
+    function setDestZnsPortal(address newAddress) external override onlyAdmin {
+        if (newAddress == address(0)) revert ZeroAddressPassed();
+
+        destZnsPortal = newAddress;
+
+        emit DestZnsPortalSet(newAddress);
+    }
+
+    function _processPayment(
+        string calldata label,
+        bytes32 parentHash
+    ) internal {
+        // TODO multi: do we actually want to stake on BOTH sides? What other payment option can we do ???!!!
+        // TODO multi: if we leave this option (stake as Portal address) this needs to be optimized!!!
```

Comment:
Can we not include the registrant address and have the portal use stake payment on the registrant's behalf?